api/src/main/java/ai/djl/modality/Output.java

```diff
@@ -70,4 +79,69 @@ public String getMessage() {
     public void setMessage(String message) {
         this.message = message;
     }
+
+    /**
+     * Encodes all data in the output to a binary form.
+     *
+     * @return the binary encoding
+     * @throws IOException if it fails to encode part of the data
+     */
+    @Override
+    public byte[] encode() throws IOException {
+        try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+            DataOutputStream os = new DataOutputStream(baos);
+            os.writeLong(serialVersionUID);
+            encodeInputBase(os);
+
+            os.writeInt(code);
+            os.writeUTF(message);
+
+            return baos.toByteArray();
+        }
+    }
+
+    /**
+     * Decodes the output from {@link #encode()}.
+     *
+     * @param is the data to decode from
+     * @return the decoded output
+     * @throws IOException if it fails to decode part of the output
+     */
+    public static Output decode(InputStream is) throws IOException {
+        try (DataInputStream dis = new DataInputStream(is)) {
+            if (serialVersionUID != dis.readLong()) {
+                throw new IllegalArgumentException("Invalid Input version");
+            }
+
+            Output output = new Output();
+            decodeInputBase(dis, output);
+
+            output.code = dis.readInt();
+            output.message = dis.readUTF();
+
+            return output;
+        }
+    }
+
+    /** {@inheritDoc} * */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        Output output = (Output) o;
+        return code == output.code && Objects.equals(message, output.message);
```

Comment:
We cannot just based on message and code

Reply:
It has a check based on `super.equals()` a few lines above. This is the IntelliJ equals template